src/test/node.ts

```diff
@@ -121,21 +134,29 @@ describe("node", () => {
 
 			const stream = fromNodeReadable(nodeStream);
 
-			let aborted: Error | undefined;
-			await stream
-				.forEach(
-					// simulate a failing `write` call
-					(_chunk) => Promise.reject(new Error(writeErrorMessage)),
-					// simulate a failing `end` call
-					(_endErr) => Promise.reject(new Error(endErrorMessage)),
-					(abortErr) => {
-						aborted = abortErr;
-					}
-				)
-				.catch((e) => e);
+			await new Promise<void>((resolve, reject) => {
+				stream
+					.forEach(
+						// simulate a failing `write` call.
+						(_chunk) =>
+							Promise.reject(new Error(writeErrorMessage)),
+						// simulate a failing `end` call.
+						(_endErr) => {
+							resolve();
```

Comment:
I think this is definitely not the way to go, because you're now resolving a promise as soon as the ender is called, but there's a whole other chain going to happen (waiting for the ender to be resolved, then propagating that back up the chain, then leading to the final `.result()` promise being resolved, etc.).

I suppose it's best to revert this last commit, and continue from the one before this (996a2f6ff2542f2d277e4b1b690e0c27bab6ca9f), which seems to be close to how it should be.